libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -47,8 +49,13 @@ export class BookSearchComponent {
   }
 
   searchBooks() {
-    if (this.searchForm.value.term) {
-      this.store.dispatch(searchBooks({ term: this.searchTerm }));
+    if (this.searchTerm) {
+      const searchTerm$ = from(this.searchTerm);
```

Comment:
Again, please take a look at the reactive forms API (https://angular.io/guide/reactive-forms). There's something called `valueChanges()` that could help us read the values of that input!

Reply:
Implemented valueChanges(), reduced significantly the code required to achieve the same functionality.